File: app/schemas/company.py
# DOMAIN: BACKEND
# LAST_MODIFIED: 2026-01-16 13:15:00
from pydantic import BaseModel, ConfigDict, Field
from typing import List, Optional, Any
from datetime import time, datetime
from app.schemas.core import Monetary, OptionalMonetary

class CompanyPublic(BaseModel):
    name: str
    slug: str
    is_active: bool
    logo_url: Optional[str] = None
    primary_color: str = "#ea580c"
    banner_url: Optional[str] = None
    background_color: Optional[str] = "#f9fafb"
    text_color: Optional[str] = "#111827"
    accent_color: Optional[str] = "#ea580c"
    opens_at: Optional[time] = None
    closes_at: Optional[time] = None
    owner_email: Optional[str] = None
    pix_key: Optional[str] = None
    instagram_url: Optional[str] = None
    whatsapp_number: Optional[str] = None
    wifi_ssid: Optional[str] = None
    wifi_password: Optional[str] = None
    segment: str = "gastro"
    model_config = ConfigDict(from_attributes=True)

class CompanyAdminSettings(CompanyPublic):
    mp_access_token: Optional[str] = None
    payment_provider: str = "NONE"
    marketplace_fee_percentage: OptionalMonetary = 0
    loyalty_percentage: OptionalMonetary = 0
    plan_tier: str = "free"
    trial_ends_at: Optional[datetime] = None
    stripe_subscription_id: Optional[str] = None
    subscription_status: Optional[str] = None
    cnpj: Optional[str] = None
    inscricao_estadual: Optional[str] = None
    fiscal_token: Optional[str] = None
    csc_token: Optional[str] = None
    csc_id: Optional[str] = None
    service_fee_percentage: OptionalMonetary = 10.0
    fixed_delivery_fee: OptionalMonetary = 0.0
    whatsapp_api_url: Optional[str] = None
    whatsapp_instance: Optional[str] = None
    whatsapp_token: Optional[str] = None
    ifood_merchant_id: Optional[str] = None
    # Kiosk Settings
    kiosk_password_set: bool = False # Retorna true se houver senha configurada, sem retornar o hash
    model_config = ConfigDict(from_attributes=True)

class CompanyUpdate(BaseModel):
    name: Optional[str] = None
    logo_url: Optional[str] = None
    banner_url: Optional[str] = None
    primary_color: Optional[str] = None
    background_color: Optional[str] = None
    text_color: Optional[str] = None
    accent_color: Optional[str] = None
    opens_at: Optional[time] = None
    closes_at: Optional[time] = None
    pix_key: Optional[str] = None
    mp_access_token: Optional[str] = None
    loyalty_percentage: OptionalMonetary = None
    instagram_url: Optional[str] = None
    whatsapp_number: Optional[str] = None
    whatsapp_api_url: Optional[str] = None
    whatsapp_instance: Optional[str] = None
    whatsapp_token: Optional[str] = None
    wifi_ssid: Optional[str] = None
    wifi_password: Optional[str] = None
    cnpj: Optional[str] = None
    inscricao_estadual: Optional[str] = None
    fiscal_token: Optional[str] = None
    csc_token: Optional[str] = None
    csc_id: Optional[str] = None
    service_fee_percentage: OptionalMonetary = None
    fixed_delivery_fee: OptionalMonetary = None
    ifood_merchant_id: Optional[str] = None
    ifood_token: Optional[str] = None
    # Kiosk Update
    kiosk_password: Optional[str] = None # Plain text para update

class KioskValidationRequest(BaseModel):
    password: str

File: frontend/src/types/company.ts
/**
 * Author: MESAFLOW_AI_SOVEREIGN
 * Version: 13.3.0 (Evolution API Integration)
 * DNA_ID: MF-TYPES-COMPANY-V13-3
 */
export type PlanTier = 'free' | 'pro' | 'enterprise';
export type CompanySegment = 'gastro' | 'event' | 'hotel' | 'corp';
export type PaymentProvider = 'mercadopago' | 'efi' | 'stripe' | 'pagarme' | 'none';

export interface Company {
  id: string;
  name: string;
  slug: string;
  custom_domain?: string | null;
  is_active: boolean;
  segment: CompanySegment;
  created_at: string;
  owner_email: string;
  owner_phone?: string | null;
  owner_role?: string | null;
  logo_url?: string | null;
  banner_url?: string | null;
  primary_color: string;
  background_color?: string | null;
  text_color?: string | null;
  accent_color?: string | null;
  stone_merchant_id?: string | null;
  stone_terminal_id?: string | null;
  opens_at?: string | null;
  closes_at?: string | null;
  qr_config?: {
    show_wifi: boolean;
    show_instagram: boolean;
    show_steps: boolean;
    show_logo: boolean;
    dark_mode: boolean;
    custom_color: string | null;
  };
  wifi_ssid?: string | null;
  wifi_password?: string | null;
  instagram_url?: string | null;
  whatsapp_number?: string | null;
  whatsapp_api_url?: string | null;
  whatsapp_instance?: string | null;
  whatsapp_token?: string | null;
  payment_provider: PaymentProvider;
  payment_credentials?: any;
  pix_key?: string | null;
  mp_access_token?: string | null;
  mp_user_id?: string | null;
  marketplace_fee_percentage?: number;
  service_fee_percentage?: number;
  fixed_delivery_fee?: number;
  loyalty_percentage?: number;
  stripe_customer_id?: string | null;
  stripe_subscription_id?: string | null;
  subscription_status?: string | null;
  plan_tier: PlanTier;
  cnpj?: string | null;
  inscricao_estadual?: string | null;
  fiscal_token?: string | null;
  csc_token?: string | null;
  csc_id?: string | null;
  kiosk_password_set?: boolean;
  kiosk_password?: string;
}

/**
 * Representa um registro imutável de ação no sistema.
 */
export interface AuditLog {
  id: number;
  user_name: string;
  user_role: string;
  action: string;
  resource: string;
  resource_id: string;
  details: any;
  ip_address: string;
  created_at: string;
}

/**
 * Representa uma assinatura de evento externo.
 */
export interface WebhookResponse {
  id: number;
  target_url: string;
  events: string[];
  secret: string;
  is_active: boolean;
  created_at: string;
}